src/main/java/Ui.java

```diff
@@ -0,0 +1,80 @@
+import java.util.Scanner;
+
+/**
+ * Represents the User Interface of Duke, including the user input fields.
+ */
+public class Ui {
```

Comment:
In general, I think the abstractions are well done (e.g.  showError, showGoodbye). It makes the code readable and more importantly highly extensible